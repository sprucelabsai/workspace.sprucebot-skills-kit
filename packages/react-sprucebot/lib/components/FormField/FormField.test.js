'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

require('jest-styled-components');

var _FormField = require('./FormField');

var _FormField2 = _interopRequireDefault(_FormField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var onChange = jest.fn();

test('it renders set value and triggers onChange', function () {
	var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(
		_FormField2.default,
		{
			type: 'text',
			input: { value: 'test1', onChange: onChange },
			meta: { touched: false }
		},
		_react2.default.createElement('input', null)
	));
	expect(wrapper).toMatchSnapshot();
	wrapper.find('input[type="text"]').simulate('change', { target: { value: 'test' } });
	expect(onChange).toHaveBeenCalledWith({ target: { value: 'test' } });
});

test('it renders with a label, finePrint, placeholder, and error', function () {
	var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(
		_FormField2.default,
		{
			label: 'Test Label',
			placeholder: 'Placeholdn\'',
			type: 'text',
			input: { value: 'test2', onChange: onChange },
			meta: { touched: true, error: 'TEST ERROR' },
			finePrint: 'Fine Print!'
		},
		_react2.default.createElement('input', null)
	));
	expect(wrapper).toMatchSnapshot();
});