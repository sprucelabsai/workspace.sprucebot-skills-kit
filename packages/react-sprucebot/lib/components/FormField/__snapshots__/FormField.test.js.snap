// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders set value and triggers onChange 1`] = `
<FormField__FieldWrapper>
  <input
    className=""
    onChange={[MockFunction]}
    type="text"
    value="test1"
  />
</FormField__FieldWrapper>
`;

exports[`it renders with a label, finePrint, placeholder, and error 1`] = `
<FormField__FieldWrapper>
  <FormField__FieldLabel>
    Test Label
  </FormField__FieldLabel>
  <input
    className="with_label"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": Object {
                "value": "test",
              },
            },
          ],
        ],
      }
    }
    placeholder="Placeholdn'"
    type="text"
    value="test2"
  />
  <FormField__FieldError>
    TEST ERROR
  </FormField__FieldError>
  <Typography__Paragraph
    fine={true}
  >
    Fine Print!
  </Typography__Paragraph>
</FormField__FieldWrapper>
`;
